src/guidellm/core/distribution.py

```diff
@@ -1,202 +1,159 @@
-from typing import List, Union
+from typing import List, Optional, Union
 
 import numpy as np
 from loguru import logger
 
+from guidellm.core.serializable import Serializable
+
 __all__ = ["Distribution"]
 
 
-class Distribution:
+class Distribution(Serializable):
     """
-    A class to represent a statistical distribution and perform various statistical
-    analyses.
-
-    :param data: List of numerical data points (int or float) to initialize the
-        distribution.
-    :type data: List[Union[int, float]], optional
+    A class to represent a statistical distribution and perform various
+    statistical analyses.
     """
 
-    def __init__(self, data: List[Union[int, float]] = None):
-        """
-        Initialize the Distribution with optional data.
+    data: Optional[List[Union[int, float]]] = []
```

Comment:
that one is similar to dataclasses in Python
```suggestion
    from pydantic import Field
    data: List[Union[int, float]] = Field(default_factory=list)
```

Reply:
yep, thanks Dmytro, forgot about the issue with mutable defaults. Will update across with the Field for factory creation